### ISSUE TITLE
False positive in NoWhiteSpaceAfter

### ISSUE THREAD

**Issue by DrVanScott** [checkstyle/checkstyle#12345]
hi,

starting with ~~8.45~~ 8.44, checkstyle generates a false positive for this code

```
public class Demo {

    Object o;

    void some() {
        Object oo = new Object[4];
        this.o = ((Object[]) oo)[1];
    }
}
```
This leads to message like "There is a whitespace after 'o'"

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="NoWhitespaceAfter">
    </module>
  </module>
</module>
```
produces this output using commandline / latest release 10.3.4
```
Starting audit...
[ERROR] ...\Demo.java:7:33: 'o' is followed by whitespace. [NoWhitespaceAfter]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by nrmancuso** [comment]
Please follow the issue template.

**Comment by DrVanScott** [comment]
Sorry, was wrong about the version. Corrected to 8.44

**Comment by DrVanScott** [comment]
Hi, don't really now all that information. I use Eclipse to verify my code. What else do you need? It depends on the above example, on rule "NoWhitespaceAfter" and on a version >= 8.44. Should be easy to reproduce.

**Comment by romani** [comment]
Please reproduce it by our CLI as issue template demands

**Comment by DrVanScott** [comment]
Above jave code (Demo.java) compiles without error or warning

Using this config:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="NoWhitespaceAfter">
    </module>
  </module>
</module>
```
produces this output using commandline / latest release 10.3.4
```
Starting audit...
[ERROR] ...\Demo.java:7:33: 'o' is followed by whitespace. [NoWhitespaceAfter]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by strkkk** [comment]
I reproduced issue, it is clearly a bug. I will try to submit PR with fix soon.

**Comment by nrmancuso** [comment]
Strange, it seems to be related to the combination of redundant `this` and casting to array type with access:

```

➜  src cat Test.java                                              
public class Test {

    Object o;

    void m1() {
        Object oo = new Object[4];
        this.o = ((Object[]) oo)[1]; // violation is here
    }

    void m2() {
        Object oo = new Object[4];
        o = ((Object[]) oo)[1];
    }

    void m3() {
        Object oo = new Object[4];
        this.o = new Object();
    }

    void m4() {
        Object oo = new Object[4];
        this.o = (int) oo;
    }

    void m5() {
        Object oo = new Object[4];
        this.o = ((int[][]) new Object())[1];
    }
}%                                                                                                             

 ➜  src cat config.xml                                             
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="NoWhitespaceAfter">
    </module>
  </module>
</module>%  
                                                                                                    
➜  src javac Test.java                                            
➜  src java -jar checkstyle-10.3.4-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/checkstyle_tester/src/Test.java:7:33: 'o' is followed by whitespace. [NoWhitespaceAfter]
[ERROR] /home/nick/IdeaProjects/checkstyle_tester/src/Test.java:27:42: 'o' is followed by whitespace. [NoWhitespaceAfter]
Audit done.
Checkstyle ends with 2 errors.

```

**Comment by strkkk** [comment]
Fix is merged
